### PR TITLE
Add salesforcebulk-connector-example

### PR DESCRIPTION
**Pull request by KasunAratthanage**

### CODE REVIEW

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md [hunk 1]
```diff
@@ -0,0 +1,338 @@
+# Salesforce Bulk Connector Example
+
+The Salesforce Bulk Connector allows you to access the [Salesforce Bulk REST API](https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/asynch_api_intro.htm) through WSO2 Enterprise Integrator (WSO2 EI). SalesforceBulk is a RESTful API that allows you to quickly load large sets of your organisation’s data into Salesforce or delete large sets of your organisation’s data from Salesforce. You can use SalesforceBulk to query, insert, update, upsert or delete a large number of records asynchronously, by submitting the records in batches. Salesforce can process these batches in the background.
+
+## What you'll build
+
+This example demonstrates how to use Microsoft Azure Storage connector to:
+
+1. Insert employee details (job and batch) in to the Salesforce.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:9]:
```suggestion
1. Insert employee details (job and batch) into Salesforce.
```

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+2. Get status of the inserted employee details.
+
+All two operations are exposed via an API. The API with the context `/resources` has two resources.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:12]:
```suggestion
Both operations are exposed via an API. The API with the context `/resources` has two resources.
```

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+
+* `/insertEmployeeDetails` : Creating a new job in the in the Salesforce account and insert employee details.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:14]:
```suggestion
* `/insertEmployeeDetails` : Creating a new job in the Salesforce account and insert employee details.
```

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+* `/getStatusOfBatch` : Retrieve status about the created batch from the Salesforce account.
+
+In this example the user sends the request to invoke an API to insert bulk of employee details to the salesforce account. When invoke the `insertEmployeeDetails` resource it will creates a new job based on the properties that you specify. Then read CSV data file by using the WSO2 File Connector and extracted data set insert as a batch. After it is responded according to the specified template and sent back to the client. Finally user can retrieve the batch status using `getStatusOfBatch` resource.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:17]:
```suggestion
In this example, the user sends the request to invoke an API to insert employee details in bulk to the Salesforce account. When invoking the `insertEmployeeDetails` resource, it creates a new job based on the properties that you specify. Read the CSV data file by using the WSO2 File Connector and the extracted dataset is inserted as a batch. Afterwards it responds according to the specified template and is sent back to the client. Finally a user can retrieve the batch status using the `getStatusOfBatch` resource. 
```

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+
+### Add integration logic
+
+First create an API, which will be where we configure the integration logic. Right click on the created Integration Project and select, **New** -> **Rest API** to create the REST API. Specify the API name as `Salesforcebulk-API` and API context as `/salesforce`.
+    
+<img src="../../../../assets/img/connectors/adding-an-api.png" title="Adding a Rest API" width="800" alt="Adding a Rest API"/>
+
+#### Configure a resource for the insertEmployeeBulkRecords 
+
+Now follow the steps below to add configurations to the insertEmployeeBulkRecords resource.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:41]:
```suggestion
Now follow the steps below to add configurations to the `insertEmployeeBulkRecords` resource.
```

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+    
+1. Initialize the connector.
+    
+    1. Follow these steps to [generate the Access Tokens for Salesforce](salesforcebulk-connector-configuration.md) and obtain the Client Id, Client Secret, Access Token, and Refresh Token.
+    
+    2. Navigate into the **Palette** pane and select the graphical operations icons listed under **Salesforcebulk Connector** section. Then drag and drop the `init` operation into the Design pane.
+        
+        <img src="../../../../assets/img/connectors/salesforcebulk-drag-and-drop-init.png" title="Drag and drop init operation" width="500" alt="Drag and drop init operation"/>
+        
+    3. Add the property values into the `init` operation as shown bellow. Replace the `clientSecret`, `clientId`, `accessToken`, `refreshToken` with obtained values from above steps.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:51]:
```suggestion
    3. Add the property values into the `init` operation as shown below. Replace the `clientSecret`, `clientId`, `accessToken`, `refreshToken` with obtained values from above steps.
```

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+      
+        - **clientSecret** : Value of your client secret given when you registered your application with Salesforce.
+        - **clientId** : Value of your client ID given when you registered your application with Salesforce.
+        - **accessToken** : Value of the access token to access the API via request.
+        - **refreshToken** : Value of the refresh token.
+       
+        <img src="../../../../assets/img/connectors/salesforcebulk-api-init-operation-parameters.png" title="Add values to the init operation" width="800" alt="Add values to the init operation"/>
+     
+2. Set up the createJob operation.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:60]:
```suggestion
2. Set up the `createJob` operation.
```

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+
+    1. Setup the `createJob` configurations. In this operation we are going to create a job in the Salesforce account. Please find the `createJob` operation parameters listed here.
+       
+        - **operation** : The processing operation that the job should perform.
+        - **object** : The object type of data that is to be processed by the job.
+        - **contentType** : The content type of the job.
+        
+        While invoking the API, the above `object` parameter value come as a user input.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:68]:
```suggestion
        While invoking the API, the above `object` parameter value comes as a user input.
```

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md [hunk 9]
```diff
@@ -0,0 +1,338 @@
...
+    
+    2. Navigate into the **Palette** pane and select the graphical operations icons listed under **Salesforcebulk Connector** section. Then drag and drop the `createJob` operation into the Design pane.
+    
+        <img src="../../../../assets/img/connectors/salesforcebulk-drag-and-drop-createjob.png" title="Drag and drop creatJobe operation" width="500" alt="Drag and drop createJob operations"/>
+    
+    3. To get the input values in to the API we can use the [property mediator](../../../mediators/property-Mediator). Navigate into the **Palette** pane and select the graphical mediators icons listed under **Mediators** section. Then drag and drop the `Property` mediators into the Design pane as shown bellow.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:74]:
```suggestion
    3. To get the input values into the API, we can use the [property mediator](../../../mediators/property-Mediator). Navigate into the **Palette** pane and select the graphical mediators icons listed under **Mediators** section. Then drag and drop the `Property` mediators into the Design pane as shown below.
```

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md [hunk 10]
```diff
@@ -0,0 +1,338 @@
...
+    
+        <img src="../../../../assets/img/connectors/salesforcebulk-api-drag-and-drop-property-mediator.png" title="Add property mediators" width="800" alt="Add property mediators"/>
+
+        The parameters available for configuring the Property mediator are as follows:
+    
+        > **Note**: That the properties should be add to the pallet before create the operation.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:80]:
```suggestion
        > **Note**: The properties should be added to the palette before creating the operation.
```

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md [hunk 11]
```diff
@@ -0,0 +1,338 @@
...
+    
+    4. Add the property mediator to capture the `objectName` value.  The object type of data that is to be processed by the job.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:82]:
```suggestion
    4. Add the property mediator to capture the `objectName` value.  This is the object type of data that is to be processed by the job.
```

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md (hunk 1 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:9]: Updated.

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md (hunk 2 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:12]: Updated.

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md (hunk 3 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:14]: Updated.

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md (hunk 4 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:17]: Updated.

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md (hunk 5 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:41]: Updated.

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md (hunk 6 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:51]: Updated.

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md (hunk 7 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:60]: Updated.

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md (hunk 8 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:68]: Updated.

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md (hunk 9 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:74]: Updated.

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md (hunk 10 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:80]: Updated.

File: en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md (hunk 11 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/salesforcebulk-connector/salesforcebulk-connector-example.md:82]: Updated.

**Pull request merged**